src/webhook/mutation/pod_mutator/oneagent_mutation/env_test.go

```diff
@@ -11,16 +11,80 @@ import (
 )
 
 func TestAddPreloadEnv(t *testing.T) {
```

Comment:
Candidate for a table test, in the current state there's a lot of duplicate code.

Reply:
IMO, table test would be to much hassle, and I am still not a fan of them. 😅 

But yeah, will look into minimising the code duplications 😉

Reply:
I tried the table tests and they actually add 3 lines more code 😄 (defining the struct for the testCases, is 4 lines so we would technically save 1 line)

and they feel way more brittle, so I will not do it :D
here is how it would have looked like:
```
type preloadTestCase struct {
	name         string
	existingPath string
	expectedPath string
}

func TestAddPreloadEnv(t *testing.T) {
	installPath := "path/test"
	testCases := []preloadTestCase{
		{
			name: "Add preload env",
			existingPath: "",
			expectedPath: installPath,
		},
		{
			name: "Concat preload env, default delimiter",
			existingPath: "path/user",
			expectedPath: "path/user" + ":" + installPath,
		},
		{
			name: "Concat preload env, respect delimiter",
			existingPath: "path1/user path2/user",
			expectedPath: "path1/user path2/user" + " " + installPath,
		},
		{
			name: "Ignore preload env, if value already present",
			existingPath: "path1/user path2/user" + " " + installPath,
			expectedPath: "path1/user path2/user" + " " + installPath,
		},
	}

	for _, test := range testCases {
		t.Run(test.name, func(t *testing.T) {
			container := createContainerWithPreloadEnv(test.existingPath)

			addPreloadEnv(container, installPath)

			verifyContainerWithPreloadEnv(t, container, test.expectedPath)
		})
	}
}
```